store/redis/redis.go

```diff
@@ -201,7 +201,7 @@ type getter func() (interface{}, error)
 // pusher defines a func type which pushes data blob into watch channel
 type pusher func(interface{})
 
-func watchLoop(msgCh chan *redis.Message, stopCh <-chan struct{}, get getter, push pusher) error {
+func watchLoop(msgCh chan *redis.Message, _ <-chan struct{}, get getter, push pusher) error {
```

Comment:
If I understand correctly and if I'm not wrong from succinctly refreshing my mind on that code, redis watch has a Receive loop and an internal `watchLoop`, both are launched in a goroutine. The Receive loop makes use of the `stopCh` but not the internal `watchLoop`. Thus if a client wants to stop the Watch using the channel it will stop the receive loop correctly but leak the main watch goroutine. Making this a wildcard will only make us think that the `stopCh` is not necessary in this method while it actually is.

I would simply do like other store drivers and add that piece in the `for := range` loop below:

```go
// Check if the watch was stopped by the caller
select {
	case <-stopCh:
		return
	default:
}
```

Reply:
Ok, I understand that portion better now after a small re-read. `receiveLoop` defers the close of `redis.Message` channel upon receiving to `stopCh`, channel which is consumed by `watchLoop`. Thus it will stop both loops and not leak. Failed to see that during the review.